src/components/address-bar/container.tsx

```diff
@@ -36,7 +38,17 @@ export class Container extends React.Component<Properties> {
   }
 
   static mapActions(_props: Properties): Partial<Properties> {
-    return {};
+    return { setDeepestVisitedRoute };
+  }
+
+  componentDidUpdate(prevProps: Properties) {
+    const {
+      app: { type },
+      route,
+    } = this.props;
+    if (prevProps.app.type !== type && prevProps.route !== route) {
+      this.props.setDeepestVisitedRoute(route);
```

Comment:
i don't think we should be setting this from this component. because this is essentially part of a state machine around the routing, if we enable setting it from arbitrary components it can get out of sync. there should really only be a single place that this is modified, and the component that is currently responsible for doing that is `src/zns-route-connect.tsx`. i think this task is likely going to require some refactoring of that component, as well as potentially some sagas, to ensure that this state is always mirroring what has happened at the level of the router.

let me know if you're running into issues. i can look a little closer at what this will need, if you'd like, or we can pair on it.